src/store/modules/DownloadModule.ts

```diff
@@ -57,6 +59,13 @@ export const DownloadModule = {
             settings.setIgnoreCache(!settings.getContext().global.ignoreCache);
             commit('setIgnoreCacheVuexOnly', settings.getContext().global.ignoreCache);
         },
+        async retryDownload({commit, rootGetters}, download: DownloadProgress) {
+            commit('removeDownload', download);
+
+            download.initialMods.forEach(combo => {
```

Comment:
Clever to call the static method here.

However, having to call it separately for each combo is IMO a problem. When using "update all" for a profile with hundred mods pending update, this creates hundred new rows into the DownloadMonitor's list. While this works it makes the UI quite unresponsive, the downloads takes ages as there's so many simultaneous ones, and the installation takes ages as each download requests and waits for file lock to update the mods.yml.

Although I've no clear path to it yet, I think we can approach this problem smarter if we do some other refactoring first. We can discuss them in voice chat. So I'd suggest we leave this task on hold for now.